src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double RESIZE_FACTOR = 1.5;
+    private static final int ELEMENT_NOT_FOUND = -1;
+    private int size;
+    private Object[] array = new Object[CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for size " + size);
+        }
+        growIfArrayFull();
 
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() + size > array.length) {
+            while (list.size() + size > array.length) {
+                resize();
+            }
+        }
+        for (int i = 0; i < list.size(); i++) {
+            array[size + i] = list.get(i);
+        }
+        size += list.size();
     }
```

Comment:
The resize logic is complicated and repeated in the `addAll(List<T> list)` method. It would be better to encapsulate the resizing logic in the `resize()` method and call it directly instead of using a while loop to ensure the capacity.